kernel/os/include/os/arch/pic32/os/os_arch.h

```diff
@@ -59,7 +59,7 @@ typedef uint32_t os_stack_t;
 /* Exit a critical section, restore processor state and unblock interrupts */
 #define OS_EXIT_CRITICAL(__os_sr) __builtin_set_isr_state(__os_sr)
 /*  This is not the only way interrupts can be disabled */
-#define OS_IS_CRITICAL() ((__builtin_get_isr_state() & 1) == 0)
+#define OS_IS_CRITICAL() ((_CP0_GET_STATUS() & 1) == 0)
```

Comment:
Could you put this change in another commit ? From my understanding, the only difference is that ```__builtin_get_isr_state``` adds an extra ```& 0xF``` to CP0 status register while you only want to check EI bit. Is that correct ?